### DISCUSSION TITLE
Each Cell contains  a Row property and each Row contains a Cells array. Is that causing endless recursion?

### DISCUSSION THREAD

**Discussion by korbibaum** [TanStack/table#2857] [Q&A]
Hey, I have a basic react-table and noticed this behaviour:
When I inspect the row property of a cell in the console, I can see an endless recursive nested arrays and objects. 

`allCells: Array<Cell>` is an array of cell objects and each cell object contains an `row: Row` which is an array of cells. **Is this how it's supposed to be?** And why is that?
[https://github.com/tannerlinsley/react-table/blob/e0b28618b201f7d76215a0085f8296a1db529761/docs/src/pages/docs/api/useTable.md#row-properties](url)

<img width="1354" alt="Screenshot 2020-11-15 at 12 16 13" src="https://user-images.githubusercontent.com/32970755/99183477-5e630200-273c-11eb-8ea5-1974147f1882.png">

**Comment by tannerlinsley** [comment]
Yep, this is purposeful. For convenience in API usage.

**Reply by korbibaum** [reply]
Thank you for the quick answer!
Doesn't it cause performance issues when passing a row prop which has endless recursive arrays in it? Where does the recursion stop?
